source/Meadow-Filter-Parse.js

```diff
@@ -140,6 +265,63 @@ const addFilterStanzaToQuery = (pFilterStanza, pQuery) =>
 			pQuery.addSort({ Column: pFilterStanza.Field, Direction: tmpSortDirection });
 			break;
 
+		case 'FBJV':   // Filter by JSON Value (left-side AND)
+		{
+			const { fieldColumn, jsonPath } = parseJSONFieldAndPath(pFilterStanza);
+			addFilterJSONToQuery(fieldColumn, jsonPath, pFilterStanza.Value, getFilterComparisonOperator(pFilterStanza.Operator), 'AND', pQuery);
+			break;
+		}
+
+		case 'FBJVOR': // Filter by JSON Value (left-side OR)
+		{
+			const { fieldColumn, jsonPath } = parseJSONFieldAndPath(pFilterStanza);
+			addFilterJSONToQuery(fieldColumn, jsonPath, pFilterStanza.Value, getFilterComparisonOperator(pFilterStanza.Operator), 'OR', pQuery);
+			break;
+		}
+
+		case 'FBJL':   // Filter by JSON List (left-side AND)
+		{
+			const { fieldColumn, jsonPath } = parseJSONFieldAndPath(pFilterStanza);
+			// Just split the value by comma for now.  May want to revisit better characters or techniques later.
+			addFilterJSONToQuery(fieldColumn, jsonPath, pFilterStanza.Value.split(','), getFilterComparisonOperator(pFilterStanza.Operator), 'AND', pQuery);
+			break;
+		}
+
+		case 'FBJLOR': // Filter by JSON List (left-side OR)
+		{
+			const { fieldColumn, jsonPath } = parseJSONFieldAndPath(pFilterStanza);
+			// Just split the value by comma for now.  May want to revisit better characters or techniques later.
+			addFilterJSONToQuery(fieldColumn, jsonPath, pFilterStanza.Value.split(','), getFilterComparisonOperator(pFilterStanza.Operator), 'OR', pQuery);
+			break;
+		}
+
+		case 'FBJD':   // Filter by JSON Date (exclude time)
+		{
+			const { fieldColumn, jsonPath } = parseJSONFieldAndPath(pFilterStanza);
+			pQuery.addFilter('', '', '(');
+			// NOTE: bypassing the table name injection by not providing a column argument
+			// TODO: prefix 'fieldColumn' with table name to avoid ambiguity (e.g. in joins)
+			pQuery.addFilter('', 1, `JSON_VALID(${fieldColumn}) =`, 'AND', 'validJson');
```

Comment:
Is this broken in foxhound, or is this caused by the API code here? https://github.com/paviasystems/mpi_api/blob/master/ts_source/controller/HeadlightController.ts#L555-L566

If the latter, it seems like we could just fix this there, and not need this hack.

Reply:
oh! I think it is. Nice, will look into cleaning that up